### PR TITLE
feat: add sign in and MFA verification

### PR DESCRIPTION
**Pull request by karolsojko**
I know this is a big PR but turns out the "sign in" feature in syncing server ruby is quite a big feature that branches out across many scopes. I replicated the mechanism in here with a bit of refactor. Instead of using the omnipotent user manager I'm using reusable use cases like `SignIn` or `VerifyMFA` 

The lock counters for users trying too many attempts will be implemented in a next PR to not make this one even bigger.

### CODE REVIEW

File: src/Domain/Session/SessionService.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
 import * as crypto from 'crypto'
 import * as winston from 'winston'
+import * as moment from 'moment'
```
Comment (atmoio) [inline src/Domain/Session/SessionService.ts]: Moment seems to be deprecated: https://github.com/moment/moment/

> Moment.js is a legacy project, now in maintenance mode. In most cases, you should choose a different library.

Comment (karolsojko) [reply src/Domain/Session/SessionService.ts]: oh wow .. I feel so old :'D will replace

File: src/Domain/UseCase/SignIn.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    authResponseFactory = {} as jest.Mocked<AuthResponseFactoryInterface>
+    authResponseFactory.createSuccessAuthResponse = jest.fn().mockReturnValue(authResponse)
+  })
+
+  it('should sign in a user', async () => {
+    expect(await createUseCase().execute({
+      email: 'test@test.te',
+      password: 'qweqwe123123',
+      userAgent: 'Google Chrome',
+      apiVersion: '004'
```
Comment (atmoio) [inline src/Domain/UseCase/SignIn.spec.ts]: apiVersion does not correspond to 004, 003, etc, but 20190520, 20190520, 20161215. Not sure if this is just a dummy value in this test.

Comment (karolsojko) [reply src/Domain/UseCase/SignIn.spec.ts]: yeah, it's a dummy version - I'll replace it so it doesn't induce confusion

**Review by atmoio — APPROVED** [review]

**Pull request merged**